avocado/utils/pci.py

```diff
@@ -0,0 +1,291 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+#
+# Author: Kleber Sacilotto de Souza <klebers@linux.vnet.ibm.com>
+# Author: Daniel Kreling <kreling@linux.vnet.ibm.com>
+# for get_memory_address() and get_mask()
+
+"""
+Module for all PCI devices related functions.
+"""
+
+
+import re
+import os
+from . import process
+
+
+def get_domains():
+    """
+    Gets all pci domains.
+
+    :return: List of pci domains.
+    """
+    cmd = "lspci -D | awk -F: {'print $1'} | uniq"
+    domains = process.system_output(cmd, shell=True, ignore_status=True)
+    if domains:
+        return domains.splitlines()
+
+
+def get_num_devices_in_domain(domain):
+    """
+    Gets number of devices in a pci domain.
+
+    :parm domain: pci domain.
+
+    :return: number of devices in a pci domain.
+    """
+    cmd = "ls -l /sys/class/*/ -1 | grep -i %s | wc -l" % domain
+    num_devices = process.system_output(cmd, shell=True, ignore_status=True)
+    return num_devices
+
+
+def get_disks_in_device(device):
+    """
+    Gets disks in a pci device.
+
+    :parm device: pci device.
+
+    :return: list of disks in a pci device.
+    """
+    cmd = "ls -l /dev/disk/by-path/ | grep %s | awk '{print $NF}'" % device
+    devices = process.system_output(cmd, shell=True, ignore_status=True)
+    devices_list = []
+    if devices:
+        for dev in devices.splitlines():
+            cmd = "realpath -s /dev/disk/by-path/%s" % dev
+            dev = process.system_output(cmd, shell=True, ignore_status=True)
+            devices_list.append(dev.strip('\n'))
+    return devices_list
+
+
+def get_interfaces_in_device(device):
+    """
+    Gets ethernet interface in a pci device.
+
+    :parm device: pci device.
+
+    :return: list of interfaces in a pci device.
+    """
+    cmd = "ls -l /sys/class/net/ | grep %s | awk '{print $NF}'" % device
+    devices = process.system_output(cmd, shell=True, ignore_status=True)
+    interfaces_list = []
+    if devices:
+        for dev in devices.splitlines():
+            interfaces_list.append(dev.split("/")[-1])
+    return interfaces_list
+
+
+def get_pci_fun_list(device):
+    """
+    Gets list of functions in the given pci device.
+    Example: in device 0000:03:00, functions are 0000:03:00.0 and 0000:03:00.1
+
+    :parm device: pci device.
+
+    :return: list of functions in a pci device.
+    """
+    cmd = "lspci -D | grep %s | awk {'print $1'}" % device
+    functions = process.system_output(cmd, shell=True, ignore_status=True)
+    if functions:
+        return functions.splitlines()
+
+
+def get_slot_from_sysfs(device):
+    """
+    Gets the pci slot of given device.
+
+    :parm device: pci device.
+
+    :return: slot of pci device from sysfs.
+    """
+    if not os.path.isdir('/sys/bus/pci/devices/%s' % device):
+        return
+    cmd = "cat /sys/bus/pci/devices/%s/devspec ; echo" % device
```

Comment:
Well you check for `device` directory presence, not the `devspec` file. Could you also please include the info about necessary module/kernel option for this method in docstring? It could save people some googling.